File: poo/java/Interfaces/Carbono/Building.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class Building implements CarbonFootprint{

    /* Atributos */
    private int pessoas;
    private boolean renovavel;
    private int lampadas;
    private boolean arCondicionado;

    /* Construtor */
    public Building(int pessoas, boolean renovavel,
                    int lampadas, boolean arCondicionado){
        this.pessoas = pessoas;
        this.renovavel = renovavel;
        this.lampadas = lampadas;
        this.arCondicionado = arCondicionado;
    }
    public Building(){
        this.pessoas = 5;
        this.renovavel = false;
        this.lampadas = 15;
        this.arCondicionado = false;
    }

    /* Implementacao da interface */
    public double getCarbonFootprint(){
        int limpo = 0;
        int ar = 0;

        if(this.renovavel) limpo = 1;
        if(this.arCondicionado) ar = 1;

        return (this.pessoas*0.6 + this.pessoas*ar*0.2 +
                this.lampadas*0.1 - limpo*0.5);
    }

    public String toString(){
        return "\nTipo: Building\nQuantidade de pessoas: " + this.pessoas + "\nEnergia renovavel: " +
                this.renovavel + "\nQuantidade de lampadas: " + this.lampadas + "\nAr condicionado: "+
                this.arCondicionado + "\nPegada de Carbono: " + this.getCarbonFootprint();
    }
}

File: poo/python/Interfaces/Carbono/Building.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

class Building():

    # Atributos #
    __pessoas = None
    __renovavel = None
    __lampadas = None
    __arCondicionado = None

    # Construtor #
    def __init__(self, pessoas=5, renovavel=False,
                 lampadas=15, arCondicionado=False):
        self.__pessoas = pessoas
        self.__renovavel = renovavel
        self.__lampadas = lampadas
        self.__arCondicionado = arCondicionado

    def getCarbonFootprint(self):
        return (self.__pessoas * 0.6 +
                self.__pessoas * self.__arCondicionado * 0.2 +
                self.__lampadas * 0.1 -
                self.__renovavel * 0.5)

    def __str__(self):
        return '''
                Tipo: Building
                Quantidade de pessoas: {0}
                Energia renovavel: {1}
                Quantidade de lampadas: {2}
                Ar condicionado: {3}
                Pegada de carbono: {4}'''.format(self.__pessoas,
                                                 self.__renovavel,
                                                 self.__lampadas,
                                                 self.__arCondicionado,
                                                 self.getCarbonFootprint())

    def __repr__(self):
        return self.__str__()
